safehttp/plugins/xsrf/xsrf_test.go

```diff
@@ -30,30 +31,25 @@ func (userIdentifier) UserID(r *safehttp.IncomingRequest) (string, error) {
 	return "1234", nil
 }
 
-func TestXSRFTokenPost(t *testing.T) {
-	tests := []struct {
-		name       string
-		host       string
-		path       string
-		userID     string
-		wantStatus int
-		wantHeader map[string][]string
-		wantBody   string
+var (
+	stateChangingMethods = []string{safehttp.MethodPost, safehttp.MethodPatch}
+	formTokenTests       = []struct {
+		name, userID, actionID, wantBody string
+		wantStatus                       int
+		wantHeader                       map[string][]string
 	}{
 		{
 			name:       "Valid token",
 			userID:     "1234",
-			host:       "foo.com",
-			path:       "/pizza",
+			actionID:   "https://foo.com/pizza",
 			wantStatus: 200,
 			wantHeader: map[string][]string{},
 			wantBody:   "",
 		},
 		{
-			name:       "Invalid host in token generation",
+			name:       "Token mismatch for invalid host",
 			userID:     "1234",
-			host:       "bar.com",
-			path:       "/pizza",
+			actionID:   "https://bar.com/pizza",
```

Comment:
Since `actionID` is just a string you could have just one test for "invalid actionID".
It feels like testing for invalid host, path and scheme is a little to redundant.